### PR TITLE
[Website] Update info on site

### LINKED ISSUE #92: Add missing robots.txt

### PR DESCRIPTION
**Pull request by MattIPv4**
Originally part of #262, all information related changes:

 - Update stats on the about page
 - Add myself under the community team to about page
 - Ensure brand name consistency
 - Ensure brand colour consistency
 - Add robots.txt with sitemap (fixes #92)

### CODE REVIEW

**Comment by MattIPv4** [comment]
Once this is merged I will begin work on the next subset of #262 so that I can resolve conflicts as I go instead of after (mainly due to massive file refactoring occurring).

**Comment by MattIPv4** [comment]
Putting this as WIP whilst I wait for Feb stats

**Comment by MattIPv4** [comment]
Latest stats added and conflicts resolved. Ready to go now @PeterDaveHello

**Comment by PeterDaveHello** [comment]
@MattIPv4 can you please remove `Merge branch 'master' into MattIPv4-update-info` here? This branch should rebase on the latest master branch, without other merge commits, thanks.

**Comment by MattIPv4** [comment]
@PeterDaveHello Git won't allow me to rebase due to unstaged files generated in the sri folder?

**Comment by PeterDaveHello** [comment]
@MattIPv4 you can delete those files or use sparseCheckout to not checkout those files.

**Comment by MattIPv4** [comment]
@PeterDaveHello Based on what has just been pushed, I feel rebase really isn't working due to the conflicts..?

**Comment by PeterDaveHello** [comment]
@MattIPv4 the conflicts need to be resolved no matter using rebase or merge, because they are just there.

**Comment by PeterDaveHello** [comment]
@MattIPv4 I thought the conflicts may come from the duplicated commits, commits from other authors than you should be removed ;)

**Comment by MattIPv4** [comment]
@PeterDaveHello As before, conflicts have been resolved and merged.

**Comment by PeterDaveHello** [comment]
@MattIPv4 I don't understand, the "merge commit" is still there...

**Comment by MattIPv4** [comment]
@PeterDaveHello yes, that merge commit is the commit that resolves the conflicts?

**Comment by PeterDaveHello** [comment]
@MattIPv4 that's not the preferred way to "resolve" the conflicts, that means this branch its own change is not clean

**Comment by MattIPv4** [comment]
@PeterDaveHello yes, the issue is that we track sitemap.xml which is automatically generated. This branch makes a change to the script that generates sitemap.xml and so there is a conflict. That's why the merge commit is needed to resolve that?

**Comment by PeterDaveHello** [comment]
@MattIPv4 if so, you can remove the commit and generate new sitemap again, or jutst regenerate it when git says there are conflcts.

**Comment by MattIPv4** [comment]
@PeterDaveHello I'm not sure how it generates a new sitemap?
I'm running the update.js script which is running fine but it seems to rely on public/packages.min.json of which I must have an older version?

**Comment by PeterDaveHello** [comment]
@MattIPv4 I think maybe you can simply drop the changes on the sitemap?

**Comment by MattIPv4** [comment]
@PeterDaveHello I honestly still have no idea how to rebase this from command line?

**Comment by PeterDaveHello** [comment]
@MattIPv4 Try git rebase -i HEAD~13 ?

**Comment by MattIPv4** [comment]
@PeterDaveHello `Successfully rebased and updated refs/heads/MattIPv4-update-info.` Nothing to commit or push?

**Comment by PeterDaveHello** [comment]
@MattIPv4 Maybe you didn't rebase anything in the interactive interface?

**Comment by PeterDaveHello** [comment]
@MattIPv4 Try this one: https://hackernoon.com/beginners-guide-to-interactive-rebasing-346a3f9c3a6d

**Comment by PeterDaveHello** [comment]
@PeterDaveHello or maybe just consider to commit everything from scratch without touching something not related, then you'll have a clean branch

**Comment by MattIPv4** [comment]
@PeterDaveHello I'm not sure what the issue is as to why it can't clean merge if you look at the conflict, it's where you have updated the body, which I haven't touched?

**Comment by PeterDaveHello** [comment]
@MattIPv4 nope, it's clear that you touched the sitemap, let me point it out in the diff for you

File: public/sitemap.xml [hunk 1]
```diff
@@ -1 +1 @@
...c></url><url><loc>https://cdnjs.com/libraries/flat-ui</loc></url><url><loc>https://cdnjs.com/libraries/flatpickr</loc></url><url><loc>https://cdnjs.com/libraries/flex-layout-attribute</loc></url><url><loc>https://cdnjs.com/libraries/flexboxgrid</loc></url><url><loc>https://cdnjs.com/libraries/flexibility</loc></url><url><loc>https://cdnjs.com/libraries/flexiblegs-css</loc></url><url><loc>https://cdnjs.com/libraries/flexie</loc></url><url><loc>https://cdnjs.com/libraries/flexisel</loc></url><url><loc>https://cdnjs.com/libraries/flexMenu</loc></url><url><loc>https://cdnjs.com/libraries/flexslider</loc></url><url><loc>https://cdnjs.com/libraries/flickity</loc></url><url><loc>https://cdnjs.com/libraries/flight</loc></url><url><loc>https://cdnjs.com/libraries/flipclock</loc></url><url><loc>https://cdnjs.com/libraries/flipcountdown</loc></url><url><loc>https://cdnjs.com/libraries/flipCounter</loc></url><url><loc>https://cdnjs.com/libraries/FlipDiv</loc></url><url><loc>https://cdnjs.com/libraries/float-label-css</loc></url><url><loc>https://cdnjs.com/libraries/floating.js</loc></url><url><loc>https://cdnjs.com/libraries/floatlabels.js</loc></url><url><loc>https://cdnjs.com/libraries/float
```
Comment (PeterDaveHello) [inline public/sitemap.xml]: here

**Comment by PeterDaveHello** [comment]
@PeterDaveHello looks it's cleared now!? 👍

**Comment by PeterDaveHello** [comment]
@MattIPv4 please note that there are still conflicts in templates/layout.html

**Comment by MattIPv4** [comment]
@PeterDaveHello yeah, I managed to locate all the commits that changed it. What should be done about layout?

**Comment by PeterDaveHello** [comment]
@MattIPv4 GitHub tells me that, I haven't took a look... but I thought you can try merging locally with master and you'll see the error?

**Comment by MattIPv4** [comment]
@PeterDaveHello I think by some miracle of random git commands, it is now happy! 🎉

**Comment by PeterDaveHello** [comment]
@MattIPv4 well, I'm worried about that mirable a little bit, as we better know what cause the issues so we can prvent them in the future. Anyway, I'll take a look at the new commits, thanks.

**Comment by MattIPv4** [comment]
@PeterDaveHello I'm honestly not really sure, Git seemed to think the `<body>` line had been modified by me and so it was conflicting with the updated line in master, however, I had only modified the line below. Rebasing onto master (manually resolving the conflict) did fix it though looking at my command history. Thank you for helping :)

File: templates/home.html [hunk 2]
```diff
@@ -1,51 +1,77 @@
...
+        <li><a href="https://w3techs.com/blog/entry/web_technologies_of_the_year_2018" target="_blank">
+          <i class="fa fa-check-circle"></i> The top CDN for three years</a></li>
+      </ul>
+    </div>
+  </div>
   <div class="row">
     <div class="col-md-12">
-        <input id="search-box" placeholder="Search from {{libCount}} libraries with {{libVerCount}} different versions ..." type="text" class="form-control col-md-7 search" aria-controls="example">
-        <div class="text-right text-muted">Realtime search by <a href="https://www.algolia.com/?utm_source=cdnjs&amp;utm_medium=link&amp;utm_campaign=cdnjs_search" target="_blank"><img src="/img/algolia.svg" style="height: 20px; width:75;" /></a></div>
+      <input id="search-box"
```
Comment (PeterDaveHello) [inline templates/home.html]: Can we split styling and content changing into different pull requests, please?

File: templates/home.html [hunk 3]
```diff
@@ -1,51 +1,77 @@
...
-    disable_css: 'true'
-  });
-})();
+  (function () {
+    _bsa.init('default', 'CVAI653I', 'placement:demo', {
+      target: '#default_demo',
+      align: 'horizontal',
+      disable_css: 'true'
+    });
+  })();
```
Comment (PeterDaveHello) [inline templates/home.html]: Same here, and the others like this.

File: templates/home.html [hunk 4]
```diff
@@ -1,51 +1,77 @@
...
-    "target": "https://cdnjs.com/#q={search_term_string}",
-    "query-input": "required name=search_term_string"
+  {
+    "@context": "https://schema.org",
+    "@type": "WebSite",
+    "url": "https://cdnjs.com/",
+    "potentialAction": {
+      "@type": "SearchAction",
+      "target": "https://cdnjs.com/#q={search_term_string}",
+      "query-input": "required name=search_term_string"
```
Comment (PeterDaveHello) [inline templates/home.html]: same here

**Comment by PeterDaveHello** [comment]
BTW, CDNJS -> cdnjs seems to be kind of rebranding, any reason why about that? I thought it's not really an important task right now, but just curious about it, actually, many many users don't even know the difference between cdnjs and Cloudflare, or think it's cdn.js

**Comment by MattIPv4** [comment]
@PeterDaveHello With regards to CDNJS -> cdnjs, almost everywhere but certain bits of the site we are marketed as cdnjs, so as part of updating the info on the site, I enforced consistency in the naming.

**Comment by MattIPv4** [comment]
> I thought [b81f2e3](https://github.com/cdnjs/new-website/commit/b81f2e3760df37dc9c6780dfb3257f0359dfd961) [20e5e5f](https://github.com/cdnjs/new-website/commit/20e5e5fc1fe7275128a4368ad846c5b0e28eeaa5) [3a8e72d](https://github.com/cdnjs/new-website/commit/3a8e72ddb09ef3b4d55e0581da03add68d77569a) should be squashed into a single commit as it's a single thing?

Sorry, don't know how to do this correctly?

**Comment by PeterDaveHello** [comment]
@MattIPv4 https://hackernoon.com/beginners-guide-to-interactive-rebasing-346a3f9c3a6d
use rebase with squash/fixup, you can move these commits in order and merge them into single one

**Comment by MattIPv4** [comment]
@PeterDaveHello I think I have fixed everything now?

**Comment by PeterDaveHello** [comment]
@MattIPv4 Thanks, I'll take a look ;)

File: templates/about.html [hunk 5]
```diff
@@ -4,7 +4,8 @@
       <p style="margin-top: 6px;">
 
         Everyone loves the&nbsp;<a target="_blank" href="https://developers.google.com/speed/libraries/?csw=1">Google
-        CDN</a>&nbsp;right? Even&nbsp;<a href="https://www.asp.net/ajax/cdn" target="_blank">Microsoft runs a public CDN</a>.
+        CDN</a>&nbsp;right? Even&nbsp;<a href="https://www.asp.net/ajax/cdn" target="_blank">Microsoft runs a public
+        CDN</a>.
```
Comment (PeterDaveHello) [inline templates/about.html]: this is not related to the commit topic

**Comment by PeterDaveHello** [comment]
Commit 3d9266a8ef9da7ba4e7f72d8757a39b726894655 says: `refactor about`, need to be more precise with more details.

File: templates/about.html (hunk 5 above)
Comment (MattIPv4) [reply templates/about.html]: This is due to the reformat that I performed on this page before I made major modifications to ensure code quality was maintained in my edits.

**Comment by MattIPv4** [comment]
@PeterDaveHello Anything further?

File: templates/about.html (hunk 5 above)
Comment (PeterDaveHello) [reply templates/about.html]: This is still not fixed. It doesn't make sense to be contained in this commit:
```
commit c4fc4a8dc4ca88315df7cf9aa2a071424ee59bdf
Author: MattIPv4 <matthew@cowley.org.uk>
Date:   Wed Jan 2 20:29:07 2019 +0000

    add MattIPv4 to about
```

Comment (MattIPv4) [reply templates/about.html]: Unsure why is it within this commit, will move it to the refactor commit.

**Comment by PeterDaveHello** [comment]
As a note of the looking part(I only reviewed the commit and the codes before),

The original look:
image

The new look:
image

It suddenly makes the homepage being a little bit complex, I'm not sure if it's good ... prefer to take a while to think about it, but some code refactor or about page change could be good, that's why we should separate different changes into different pull requests, so they won't block each other with unrelated reason.

**Comment by MattIPv4** [comment]
@PeterDaveHello Yeah, I'm just going to work a bit on cleaning up that design now :)

**Comment by PeterDaveHello** [comment]
@MattIPv4 thanks

**Comment by MattIPv4** [comment]
@PeterDaveHello Issues fixed, added some more padding and condensed the points on the homepage to make it cleaner. (#266 fixes the layout further in terms of responsiveness)

**Comment by MattIPv4** [comment]
@PeterDaveHello Sorry, could you possibly do the clean merge with master? I'm still not very good at doing it.

**Comment by PeterDaveHello** [comment]
@MattIPv4 you can run this command locally:
"git rebase master MattIPv4-update-info"

**Comment by MattIPv4** [comment]
@PeterDaveHello Ah, thank you! As soon as the checks complete, this should be good to go?

**Comment by MattIPv4** [comment]
cc @ryankirkman This involves a change to branding in that is consolidates the use of `cdnjs` over `CDNJS`, which appears to be how the brand is aligned currently in most places. It also enforces the use of a single, consistent brand colour.

File: templates/about.html [hunk 6]
```diff
@@ -1,113 +1,165 @@
...
+                scrolling="0" width="105px" height="20px"></iframe>
+        <iframe src="https://ghbtns.com/github-btn.html?user=cdnjs&repo=new-website&type=star&count=true"
+                frameborder="0" scrolling="0" width="105px" height="20px"></iframe>
+        <iframe src="https://ghbtns.com/github-btn.html?user=cdnjs&repo=cdnjs&type=fork&count=true" frameborder="0"
+                scrolling="0" width="105px" height="20px"></iframe>
+      </p>
       <p>
-        <img id="gravatar" src="" style="float: left; width: 64px; height: 64px; border: 1px solid #ccc; margin-right: 10px;" />Hey
-        <span style="font-weight: bold;" id="name"></span>
+        Supports <a href="https://en.wikipedia.org/wiki/HTTP/2" target="_blank"><strong>http2.0</strong>/</a>http/https/<a
```
Comment (PeterDaveHello) [inline templates/about.html]: any idea about use wiki instead of the page by IETF HTTP WG?

File: templates/about.html [hunk 7]
```diff
@@ -1,113 +1,165 @@
...
           <td><h3>Creators</h3></td>
         </tr>
-      </thead>
-      <tbody>
+        </thead>
+        <tbody>
         <tr>
           <td style="width:300px;">
             <a href="https://github.com/ryankirkman" target="_blank">
-              <img style="width: 50px; height: 50px;" src="/img/ryan.jpg" />
```
Comment (PeterDaveHello) [inline templates/about.html]: I believe the space keep it more readable, can we keep it?

File: templates/about.html (hunk 6 above)
Comment (MattIPv4) [reply templates/about.html]: This is using the wiki page, would you rather the IETF WG page? @PeterDaveHello

Comment (PeterDaveHello) [reply templates/about.html]: Just want to know more reason about change it, no mind.

File: templates/about.html [hunk 8]
```diff
@@ -1,113 +1,165 @@
...
-      <iframe src="https://ghbtns.com/github-btn.html?user=cdnjs&repo=cdnjs&type=fork&count=true" frameborder="0" scrolling="0" width="105px" height="20px"></iframe>
-      </p><p>
-      <span>Supports <a href="https://http2.github.io" target="_blank"><strong>http2.0</strong>/</a>http/https/<a href="https://en.wikipedia.org/wiki/SPDY" target="_blank">spdy</a>, we encourage always using <strong>https</strong>!
-      <br /><span>Currently, CDNJS is one of the most famous free and public web front-end CDN services which is used by ~2,999,000 websites worldwide, see more detail on <a href="https://trends.builtwith.com/cdn/CDN-JS" target="_blank">BuiltWith</a> and <a target="_blank" href="https://w3techs.com/technologies/overview/content_delivery/all">W3Techs</a>.</span>
-    </p>
-
-    <div class="logged-in" style="display: none;">
+        Everyone loves the&nbsp;<a target="_blank" href="https://developers.google.com/speed/libraries/?csw=1">Google
+        CDN</a>&nbsp;right? Even&nbsp;<a href="https://www.asp.net/ajax/cdn" target="_blank">Microsoft runs a public CDN</a>.
+        <br/>The problem is, they only host the most popular libraries.
```
Comment (PeterDaveHello) [inline templates/about.html]: same here

File: templates/about.html [hunk 9]
```diff
@@ -1,113 +1,165 @@
...
+        <ul style="list-style: none;">
+          <li>Hover over rows and click on the +/- symbols to change your favorites(yellow)</li>
+        </ul>
+      </div>
+
+      <h3>Sponsors</h3>
+      <div class="sponsors">
+
+        <a href="https://cloudflare.com/?utm_source=cdnjs&utm_medium=link&utm_campaign=cdnjs" target="_blank">
+          <img alt="cloudflare" style="margin-bottom: 20px;" src="/img/poweredbycloudflare.png"/>
```
Comment (PeterDaveHello) [inline templates/about.html]: same here

File: templates/about.html (hunk 8 above)
Comment (MattIPv4) [reply templates/about.html]: What is the issue here? @PeterDaveHello

Comment (PeterDaveHello) [reply templates/about.html]: Oh, the space ;)

```diff
- <br/>
+ <br />
```

Comment (MattIPv4) [reply templates/about.html]: @PeterDaveHello Why are we adding spaces to all self-closing tags? That seems very non-standard to me and a waste of data?

Comment (PeterDaveHello) [reply templates/about.html]: It's about readability. Just like the other spaces in indentation.

Comment (MattIPv4) [reply templates/about.html]: I don't see how it improve readability, having a space before the `/>` seems irrelevant? It just means every time someone auto-formats this code, they have to manually put all the spaces back in?

Comment (PeterDaveHello) [reply templates/about.html]: Your code formatter should be configurable to keep the space, otherwise, replace it with another one, please.

Comment (PeterDaveHello) [reply templates/about.html]: @MattIPv4 do you have any idea why we use `<br/>` but not `<br>` here?

Comment (MattIPv4) [reply templates/about.html]: @PeterDaveHello We use `<br/>` for xhtml compliancy. See https://www.w3.org/TR/xhtml1/#h-4.6

Comment (PeterDaveHello) [reply templates/about.html]: so ... https://www.w3.org/TR/xhtml1/#C_2

Comment (MattIPv4) [reply templates/about.html]: @PeterDaveHello Yup, yet their above examples in 4.6 don't have that space. I'll update webstorm to use spaces.

Comment (PeterDaveHello) [reply templates/about.html]: It's more a styling issue now, we should try to maintain the more readability styling, not only here, also the other places, like commit and commit message, and in the other languages, that's why I ask if you can add some checks in the brand repo for the python codes, this is a very important issue for collaborate.

File: templates/about.html [hunk 10]
```diff
@@ -1,113 +1,178 @@
...
+               data-show-count="false" data-lang="en">Follow @MattIPv4</a>
+          </td>
+        </tr>
+        </tbody>
+      </table>
+    </div>
   </div>
 </div>
 
-    <script async>window.twttr=(function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0],t=window.twttr||{};if(d.getElementById(id))return t;js=d.createElement(s);js.id=id;js.src="https://platform.twitter.com/widgets.js";fjs.parentNode.insertBefore(js,fjs);t._e=[];t.ready=function(f){t._e.push(f);};return t;}(document,"script","twitter-wjs"));</script>
```
Comment (PeterDaveHello) [inline templates/about.html]: I wonder if we need to touch this one as we won't even edit it?

Comment (MattIPv4) [reply templates/about.html]: Ah yes, sorry, hadn't noticed that the refactor had done that. Will rectify when I'm home later!

**Comment by MattIPv4** [comment]
| New | Old |
|------|-----|
| image | image |
| image | image |

**Comment by thomasdavis** [comment]
+1 for keeping home page minimal.

**Comment by MattIPv4** [comment]
I personally think the few key selling points that will be added here are still keeping the page minimal but also work well to help promote what cdnjs is better to a user who may not know and has ended up on the home page? @thomasdavis

**Comment by PeterDaveHello** [comment]
@MattIPv4 let's separate the homepage redesign into another PR and get this merged, what do you think?

**Comment by MattIPv4** [comment]
@PeterDaveHello Homepage design changed moved to #293 - Also, rebased against latest master.

**Comment by PeterDaveHello** [comment]
As @ryankirkman @thomasdavis @drewfreyling for review as branding changed.

**Comment by PeterDaveHello** [comment]
Never mind. @MattIPv4 let's resolve the conflicts and get it merged, thanks.

**Comment by MattIPv4** [comment]
@PeterDaveHello Thank you, rebased against latest master :)
